packages/plugin-board/src/board.js

```diff
@@ -0,0 +1,444 @@
+    // we want the first promise to resolve before continuing with the next
+    // chunk or else we'll have race conditions among patches
+    return promiseSeries(chunks.map((part) => this._addContentChunk.bind(this, conversation, channel, part)));
+  },
+
+  /**
+   * Adds Image to a Channel
+   * Uploads image to spark files and adds SCR + downloadUrl to the persistence
+   * service
+   * @memberof Board.BoardService
+   * @param  {Conversation} conversation - Contains the currently selected conversation
+   * @param  {Board~Channel} channel
+   * @param  {File} image - image to be uploaded
+   * @returns {Promise<Board~Content>}
+   */
+  addImage(conversation, channel, image) {
+    return this.spark.board._uploadImage(conversation, image)
+      .then((scr) => this.spark.board.addContent(conversation, channel, [{
+        mimeType: image.type,
+        size: image.size,
+        displayName: image.name,
+        scr
+      }]));
+  },
+
+  /**
+   * Creates a Channel
+   * @memberof Board.BoardService
+   * @param  {Board~Channel} channel
+   * @returns {Promise<Board~Channel>}
+   */
+  createChannel(channel) {
+    return this.spark.request({
+      method: `POST`,
+      api: `board`,
+      resource: `/channels`,
+      body: channel
+    })
+      .then((res) => res.body);
+  },
+
+  /**
+   * Decrypts a collection of content objects
+   *
+   * @memberof Board.BoardService
+   * @param  {Array} contents curves, text, and images
+   * @returns {Promise<Array>} Resolves with an array of {@link Board~Content} objects.
+   */
+  decryptContents(contents) {
+    return Promise.all(contents.items.map((content) => {
+      let decryptPromise;
+
+      if (content.type === `FILE`) {
+        decryptPromise = this.decryptSingleFileContent(content.payload, content.encryptionKeyUrl);
+      }
+      else {
+        decryptPromise = this.decryptSingleContent(content.payload, content.encryptionKeyUrl);
+      }
+
+      return decryptPromise
+      .then((res) => {
+        Reflect.deleteProperty(content, `payload`);
+        Reflect.deleteProperty(content, `encryptionKeyUrl`);
+        return defaults(res, content);
+      });
+    }));
+  },
+
+  /**
+   * Decryts a single STRING content object
+   * @memberof Board.BoardService
+   * @param  {string} encryptedData
+   * @param  {string} encryptionKeyUrl
+   * @returns {Promise<Board~Content>}
+   */
+  decryptSingleContent(encryptedData, encryptionKeyUrl) {
+    return this.spark.encryption.decryptText(encryptionKeyUrl, encryptedData)
+    .then((res) => JSON.parse(res));
+  },
+
+  /**
+   * Decryts a single FILE content object
+   * @memberof Board.BoardService
+   * @param  {string} encryptedData
+   * @param  {string} encryptionKeyUrl
+   * @returns {Promise<Board~Content>}
+   */
+  decryptSingleFileContent(encryptedData, encryptionKeyUrl) {
+    const payload = JSON.parse(encryptedData);
+
+    return this.spark.encryption.decryptScr(encryptionKeyUrl, payload.scr)
+    .then((scr) => {
```

Comment:
check your .then blocks. they should be indented one level above their initial line.